### DISCUSSION TITLE
Will Bitcraze produce Matrix-ToF Deck?

### DISCUSSION THREAD

**Discussion by RavenLite** [bitcraze/discussions#303] [General]
Hi, everyone

I read the blog, [Towards Autonomous Flight on Nano-Drones with Multi-zone Depth Sensors](https://www.bitcraze.io/2022/09/towards-autonomous-flight-on-nano-drones-with-multi-zone-depth-sensors/) posted on 2022-09-12, and learn about your innovative work. I'm using Multi-ranger Deck to do the mapping and planning, so I'm very interested in the Matrix-ToF Deck (or drone).
I want to ask "is there an easy way to buy them" or "does Bitcraze have plans to produce?"

Thanks!

**Comment by RavenLite** [comment]
Yes, producing hardware by ourselves and making the driver work is a headache 😭

**Comment by krichardsson** [comment]
Hi, 
we (Bitcraze) have discussed a multiranger deck with the matrix flavor of the distance sensor, but we have not decided yet what it should look like or if we will make it at all. 

The matrix sensors produce a fair amount of data which might require a more complex design with a CPU on the deck for first level processing. Another option could be to only have the matrix sensor forward and simpler sensors to the sides and back. Would that be interesting for your use case?

**Reply by RavenLite** [reply]
> Hi, we (Bitcraze) have discussed a multiranger deck with the matrix flavor of the distance sensor, but we have not decided yet what it should look like or if we will make it at all.
> 
> The matrix sensors produce a fair amount of data which might require a more complex design with a CPU on the deck for first level processing. Another option could be to only have the matrix sensor forward and simpler sensors to the sides and back. Would that be interesting for your use case?

Hi Kristoffer,
Thanks for your reply! 

For the second question (if will make it or not), I think Matrix-ToF Deck is really a good choice for Crazyflie to avoid barriers. Now, I'm using Multi-ranger Deck (VL53L1CX) to do similar things, but it will be easier to use VL53L5CX from the perspective of SDK usage. Matrix distance is a new way of thinking and I find that there is very little research on this at the moment, so I look forward to using it very much. The answer must be "**Yes**" for me 😆

For the first question (what it should look like), I prefer "only have the matrix sensor forward and simpler sensors to the sides and back". This option could keep the Matrix-ToF Deck lightweight and compatible with other decks just like now. 
Maybe for the SLAM modeling scene, the other option will be a better choice. But I guess is not a good idea to use the nano-size drone to do the complete SLAM modeling.

Please let me know if there is a timetable 🙏

**Reply by krichardsson** [reply]
Thanks for the feedback, it is very valuable to us!

Unfortunately there is no timeline, but now that we have some more information about requirements we might be able to move forward. If we do anything in this are we will announce it on the [blog](https://www.bitcraze.io/blog/)

**Reply by RavenLite** [reply]
> Thanks for the feedback, it is very valuable to us!
> 
> Unfortunately there is no timeline, but now that we have some more information about requirements we might be able to move forward. If we do anything in this are we will announce it on the [blog](https://www.bitcraze.io/blog/)

OK, I will stay concerned.
Have a good day!
